src/main/proto/streams.proto

```diff
@@ -0,0 +1,125 @@
+syntax = "proto3";
+package io.axoniq.axonserver.grpc.streams;
+import "event.proto";
+import "google/protobuf/empty.proto";
+option java_multiple_files = true;
+
+/* Service providing operations for persistent event streams, event streams where Axon Server keeps
+   track of the progress. All operations require a header (AxonIQ-Context) to be passed with each request to define
+   the context.
+ */
+service EventStreamService {
+  /* Open a persistent event stream connection from a client. Creates the stream if it does not exist. */
+  rpc OpenStream(stream StreamCommand) returns (stream StreamSignal) {
+
+  }
+
+  /* Deletes a persistent event stream. All existing connections to the stream are closed. */
+  rpc DeleteStream(DeleteStreamRequest) returns (stream google.protobuf.Empty) {
+
+  }
+
+  /* Change properties of a persistent event stream. */
+  rpc UpdateStream(UpdateStreamRequest) returns (stream google.protobuf.Empty) {
+
+  }
+
+  /* Returns a list of all persistent event streams defined (for the context). For each event stream it returns
+    the progress per segment.  */
+  rpc ListStreams( google.protobuf.Empty) returns (stream StreamStatus) {
+
+  }
+
+  /* Returns the clients connected to all persistent event streams defined (for the context). For each client it returns
+    the segments that are received by the client.  */
+  rpc ListConnections( google.protobuf.Empty) returns (stream StreamConnections) {
+
+  }
+}
+
+/* Contains the current status of a persistent stream */
+message StreamStatus {
+  string stream_id = 1; /* the unique identification of the stream */
+  string stream_name = 2; /* a name for the stream */
+  SequencingPolicy sequencing_policy = 3; /* the policy used to distribute events across segments. */
+  string filter = 4; /* an expression to filter events */
+  repeated SegmentPosition segments = 5; /* the last confirmed position per segment */
+}
+
+/* Contains the position per segment */
+message SegmentPosition {
+  int32 segment = 1; /* the segment number */
+  int64 position = 2; /* the last confirmed position */
+}
+
+/* Contains the current connections for a persistent stream */
+message StreamConnections {
+  string stream_id = 1; /* the unique identification of the stream */
+  string stream_name = 2; /* a name for the stream */
+  repeated StreamConnection segments_per_connection = 3; /* the segments held per client */
+}
+
+/* Contains the segments per client */
+message StreamConnection {
+  string client_id = 1; /* the unique identification of the client */
+  repeated int32 segments = 2; /* a list of segment numbers */
+}
+
+/* Request to set up a connection to a stream. Clients should first submit an OpenRequest on this connection
+   to connect to a stream and can then submit ProgressRequest messages to report progress.  */
+message StreamCommand {
+  oneof request {
+    OpenRequest open = 1; /* the initial message to connect to a stream */
```

Comment:
If the outer message is StreamRequest, this one could just be called "Open" or "OpenStream". Or since the description says "the initial message to connect to a stream", perhaps "Connect"?